pkg/azure/blob/credentials.go

```diff
@@ -43,6 +68,113 @@ func GetCredentials(cfg azureconfig.Config, l hclog.Logger) (CredentialSet, erro
 	return CredentialSet{Credential: *cred, Pipeline: pipe}, nil
 }
 
+func GetCredentialsFromCli(cfg azureconfig.Config, l hclog.Logger) (CredentialSet, error) {
+
+	if len(cfg.StorageAccount) == 0 || len(cfg.StorageAccountResourceGroup) == 0 {
+		return CredentialSet{}, errors.Errorf("StorageAccount and/or StorageAccountResourceGroup was not set, login with az cli not attempted")
+	}
+
+	authorizer, err := auth.NewAuthorizerFromCLI()
+	if err != nil {
+		return CredentialSet{}, errors.Errorf("Failed to login with Azure cli: %v", err)
+	}
+	subscriptionId := cfg.StorageAccountSubscriptionId
+	if len(subscriptionId) == 0 {
+		subscriptionId, err = getCurrentAzureSubscriptionFromCli()
+		if err != nil {
+			return CredentialSet{}, err
+		}
+	}
+	accountsClient := storage.NewAccountsClient(subscriptionId)
+	accountsClient.Authorizer = authorizer
+	_ = accountsClient.AddToUserAgent(UserAgent)
```

Comment:
Yeah, I did it that way to try and make it clear that it was intentional, any error here should not be terminal as its's only updating the UserAgent string, happy to handle it if you feel we should?

Reply:
I see.  Perhaps we add a comment above this line mentioning this?

Reply:
We can log a warning to to `m.Context.Err` -> `m.Err` so that if this happens we can tell that it did and track down the problem. Hiding the error would make this impossible to figure out in prod.